javascript/ql/lib/semmle/javascript/DOM.qll

```diff
@@ -465,10 +465,7 @@ module DOM {
    */
   private DataFlow::SourceNode nonFirstLocationType(DataFlow::TypeTracker t) {
     // One step inlined in the beginning.
-    exists(DataFlow::TypeTracker t2 |
-      result =
-        any(DataFlow::Node n | n.hasUnderlyingType("Location")).getALocalSource().track(t2, t)
-    )
+    result = any(DataFlow::Node n | n.hasUnderlyingType("Location")).getALocalSource().track(_, t)
```

Comment:
Actually, I think you've found a bug in our code, and I think this is a fix.

```suggestion
    result = any(DataFlow::Node n | n.hasUnderlyingType("Location")).getALocalSource().track(DataFlow::TypeTracker::start(), t)
```

Reply:
I'll have to trust you on this one 😄. Maybe this indicates we need some tests around this functionality?

Reply:
I'll look into that next year 🎄  
If you've merged this PR by then, then I'll just do another PR as a followup.

Reply:
I found a failing test. I pushed the test, and the fix, directly to this PR.